chrome/content/zotero/bibliography.js

```diff
@@ -195,20 +203,55 @@ var Zotero_File_Interface_Bibliography = new function() {
 		
 		// Change label to "Citation" or "Note" depending on style class
 		if(document.getElementById("citations")) {
+			let label = "";
 			if(Zotero.Styles.get(selectedStyle).class == "note") {
-				var label = Zotero.getString('citation.notes');
+				label = Zotero.getString('citation.notes');
 			} else {
-				var label = Zotero.getString('citation.citations');
+				label = Zotero.getString('citation.citations');
 			}
 			document.getElementById("citations").label = label;
 		}
 
 		window.sizeToContent();
+	};
+
+	/*
+	 * Update locale menulist when style is changed
+	 */
+	function updateLocaleMenu(selectedStyle) {
+		// For styles with a default-locale, disable locale menulist and show locale
+		var menulist = document.getElementById("locale-menu");
+		
+		// If not null, then menulist is extended with the default-locale value
+		// of the previously selected style
+		if (defaultStyleLocale) {
+			// Reset menulist
+			menulist.removeItemAt(0);
+			defaultStyleLocale = "";
+		}
+		
+		if (selectedStyle.locale) {
+			defaultStyleLocale = selectedStyle.locale;
+			
+			//add default-locale to menulist
+			let localeLabel = defaultStyleLocale;
+			if (Zotero.Styles.locales[defaultStyleLocale] !== undefined) {
+				localeLabel = Zotero.Styles.locales[defaultStyleLocale];
+			}
+			
+			menulist.insertItemAt(0, localeLabel, defaultStyleLocale);
+			menulist.selectedIndex = 0;
+			menulist.disabled = true;
+		} else {
+			menulist.value = selectedLocale;
+			menulist.disabled = false;
+		}
 	}
 
-	function acceptSelection() {
+	this.acceptSelection = function () {
 		// collect code
 		_io.style = document.getElementById("style-listbox").selectedItem.value;
+		_io.locale = selectedLocale;
```

Comment:
I was predicting it from code, but you can test using Scratchpad in browser context
```
var io = {};
window.openDialog("chrome://zotero/content/bibliography.xul",
			"_blank","chrome,modal,centerscreen", io);
JSON.stringify(io);
```
Select CMoS and Afrikaans, then switch to Cell and click OK. In the console window you should see `{"style":"http://www.zotero.org/styles/cell","locale":"af-ZA","mode":"bibliography","method":"copy-to-clipboard"}`